functions/pdf.js

```diff
@@ -0,0 +1,275 @@
+    doc.text(author.replace(asciRegex, ""), x, y + 7);
+  };
+  const addGreenPage = function (showText = false) {
+    doc.addPage();
+    doc.setFillColor(23, 166, 141);
+    doc.rect(0, 0, width, height, "F");
+    addBranding(logoBlack, showText);
+    usedYSpace = marginTop;
+  };
+
+  //
+
+  //   ----- FIRST PAGE
+  doc.setFillColor(23, 166, 141);
+  doc.rect(0, 0, width, height, "F");
+  addBranding();
+
+  //   Add Title
+  usedYSpace += 55;
+  addHeading(isSample ? "Your Sample" : "Your Chat", marginLeft, usedYSpace);
+
+  //    Add participants
+  usedYSpace += 10;
+  Object.keys(chat.personColorMap).forEach((key) => {
+    if (key in chat.personColorMap) {
+      if (usedYSpace + 13 > pageYSpace) {
+        addGreenPage();
+      }
+      if (key in chat.personColorMap) {
+        drawAuthorBubble(key, marginLeft, usedYSpace);
+      }
+      usedYSpace += 13;
+    }
+  });
+
+  //   new page
+  addGreenPage();
+  usedYSpace = 55;
+  //   Add fun Facts
+  addHeading("Fun Facts", marginLeft, usedYSpace);
+  usedYSpace += 10;
+
+  const funFacts = await chat.getFunFacts();
+  const funFactHeight = 40;
+  funFacts.forEach((fact) => {
+    if (fact.name in chat.personColorMap) {
+      if (usedYSpace + funFactHeight > pageYSpace) {
+        addGreenPage();
+      }
+
+      drawAuthorBubble(fact.name, marginLeft, usedYSpace);
+      doc.setFontSize(15);
+      doc.setFont("helvetica", "normal");
+
+      let factStrings = [];
+      factStrings.push("Number of Words: " + fact.numberOfWords);
+      factStrings.push("Average Message Length: " + fact.averageMessageLength);
+      factStrings.push("Unique words: " + fact.uniqueWords);
+      factStrings.push("Characters in longest Message: " + fact.longestMessage);
+
+      doc.text(factStrings, marginLeft, usedYSpace + 15);
+
+      usedYSpace += funFactHeight;
+    }
+  });
+
+  // // subtitle
+  // doc.setFontSize(16);
+  // doc.text(15, 30, "Your Chat in your hands.");
+  //   ----- Start of message pages
+  addPage();
+  doc.setTextColor(0, 0, 0);
+  doc.setFontSize(fontSize);
+  doc.setFont("helvetica", "normal");
+  const lineHeight = (fontSize * 1.5) / 3.64; // line + padding top and bottom
+
+  //   Iterate all messages
+  const messages = isSample ? chat.chatObject.slice(0, 100) : chat.chatObject;
+
+  messages.forEach((data) => {
+    let isSystem = "System" === data.author;
+    const isEgo = ego === data.author;
+
+    if (!isSystem && !(data.author in chat.personColorMap)) {
+      isSystem = true;
+    }
+
+    const splitMessage = doc.splitTextToSize(data.message, isSystem ? 120 : 60);
+    const numLines = splitMessage.length;
+    const messageHeight = lineHeight * numLines;
+
+    let messageY = calcMessageBodyHeight(numLines); // get start Y Coordinate of Message
+    let messageX = isEgo ? 58 : marginLeft + paddingMessage; // get start X Coordinate of Message
+
+    // Draw bubble
+    if (isSystem) {
+      messageX = 40;
+      doc.setFillColor(53, 53, 38);
```

Comment:
maybe we can import colors from a file, so I know what color this is?

Reply:
as with person color

Reply:
nah this is a constant color